install/kubernetes/helm/istio/templates/sidecar-injector-configmap.yaml

```diff
@@ -112,6 +114,15 @@ data:
         - name: ISTIO_META_INTERCEPTION_MODE
           value: {{ "[[ annotation .ObjectMeta `sidecar.istio.io/interceptionMode` .ProxyConfig.InterceptionMode ]]" }}
         imagePullPolicy: {{ .Values.global.imagePullPolicy }}
+        {{ "[[ if (ne (annotation .ObjectMeta `status.sidecar.istio.io/port` " }} {{ .Values.global.proxy.statusPort }} {{ ") \"0\") ]]" }}
+        readinessProbe:
+          httpGet:
+            path: {{ "[[ annotation .ObjectMeta `readiness.status.sidecar.istio.io/path` " }} "{{ .Values.global.proxy.readinessPath }}" {{ " ]]" }}
```

Comment:
Far too many options !

The path can be fixed - not reason to modify it. I think we can avoid user confusion - and just specify initialDelaySeconds default to 1, period to 1 and failureThreshold to 30
(or similar).

 The main question is what is the best value - can we look at typical startup time and see how long it takes until ADS finishes ?

Reply:
I don't think we should be in the business of saying what works for all environments. I see making it configurable as saving us a lot of pain down the road.  Choosing a "best" or even "good" value is never trivial and rarely works for everyone.